File: NumberGuessingGame.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

func main() {
    rand.Seed(time.Now().Unix())
    num := rand.Intn(100-1) + 1
    guessCount := 0
    var guess int

    for {
        fmt.Print("Guess the random number between 1 - 100: ")
        fmt.Scanf("%d", &guess)
        guessCount++

        if guess < num {
            fmt.Println("Your guess was too low, try again!")
        } else if guess > num {
            fmt.Println("Your guess was too high, try again!")
        } else if guess == num {
            break
        }
    }
    fmt.Printf("You guessed the number in %d tries\n", guessCount)
}

File: NumberGuessingGame.py
import random


num = random.randint(1, 100)
guess = 0
guessCount = 0


def main():

    global guessCount
    guess = input("Guess a number between 1-100: ")
    guessCount += 1

    if (int(guess) < int(num)):
        print("Your guess was too low, try again!")
        main()

    elif (int(guess) > int(num)):
        print("Your guess was to high, try again!")
        main()

    else:
        print("You guessed correct!")
        print("You guessed " + str(guessCount) + " number of times")
        input("Press any key to continue . . .")


main()
